inst

```diff
@@ -259,7 +262,67 @@ nextcloud_urlEncode() {
     "$1" ""
 }
 
+nextcloud_configure_oidc() {
+    if is_ucr_true appcenter/apps/nextcloud/disable-oidc-login; then
+        univention-app shell nextcloud sudo -u www-data php /var/www/html/occ app:disable user_oidc || true
+        return;
+    fi
+    if ! ucs_needsKeycloakSetup "$@"; then
+        return;
+    fi
+
+    if [ ! -e "$NC_OIDC_SECRET_FILE" ]; then
+        create_machine_password > "$NC_OIDC_SECRET_FILE"
+    fi
+
+    CLIENT_SECRET="$(< $NC_OIDC_SECRET_FILE)"
+    NC_OIDC_BASE_URL="https://$hostname.$domainname/nextcloud"
+    CLIENT_ID="$NC_OIDC_BASE_URL/"
+    univention-keycloak "$@" oidc/rp create \
+        --client-secret "$CLIENT_SECRET" \
+        --description "Nextcloud UCS App OIDC client" \
+        --name "$CLIENT_ID" \
+        --app-url "$NC_OIDC_BASE_URL" \
+        --host-fqdn "$hostname.$domainname" \
+        --redirect-uri "$NC_OIDC_BASE_URL/*" \
+        --request-uris "$NC_OIDC_BASE_URL/apps/user_oidc/*" \
+        --frontchannel-logout-url "$NC_OIDC_BASE_URL/apps/user_oidc/sls" \
+        --backchannel-logout-url "$NC_OIDC_BASE_URL/apps/user_oidc/sls" \
+        --post-logout-redirect-uris "$NC_OIDC_BASE_URL" \
+        --always-display-in-console \
+        "$CLIENT_ID" || die "Could not create Nextloud Relying Party"
+        # [--web-origins WEB_ORIGINS]
+        # [--direct-access-grants]
+        # [--allow-implicit-flow]
+        # [--access-token-lifespan ACCESS_TOKEN_LIFESPAN]
+        # [--service-accounts-enabled]
+        # [--no-frontchannel-logout]
+        # [--auth-browser-flow AUTH_BROWSER_FLOW]
+        # [--add-audience-mapper]
+        # [--access-token-audience ACCESS_TOKEN_AUDIENCE]
+        # [--id-token-audience ID_TOKEN_AUDIENCE]
+        # [--policy-url POLICY_URL]
+        # [--logo-url LOGO_URL]
+        # [--tos-url TOS_URL]
+        # [--pkce-code-challenge-method {,plain,S256}]
+        # [--default-scopes DEFAULT_SCOPES]
+        # [--optional-scopes OPTIONAL_SCOPES]
+        # [--consent CONSENT]
+
+    univention-app shell nextcloud sudo -u www-data php /var/www/html/occ app:enable user_oidc || true
+    univention-app shell nextcloud sudo -u www-data php /var/www/html/occ user_oidc:provider \
+        -c "$CLIENT_ID" \
+        -s "$CLIENT_SECRET" \
+        -d "$(univention-keycloak "$@" get-keycloak-base-url)/realms/ucs/.well-known/openid-configuration" \
+        "${appcenter_apps_nextcloud_oidc_identifier:-UCS}" || die "Could not configure Keycloak as Nextcloud OpenID Provider"
```

Comment:
@blizzz 
> One thing I noticed, OIDC does not seem to play along with LDAP, and administrators are not promited ot NC admins either.

OK, NC admins cannot login because they don't exists in Keycloak.
I didn't have problems for Administrator accounts (from UCS)

"are not promited ot " means "are not permitted or" ?